app/docker/networks/edit/NetworkDetailsView.tsx

```diff
@@ -0,0 +1,138 @@
+import { PageHeader } from '@/portainer/components/PageHeader';
+import { confirmDeletionAsync } from '@/portainer/services/modal.service/confirm';
+import { AccessControlPanel } from '@/portainer/access-control/AccessControlPanel/AccessControlPanel';
+import { ResourceControlType } from '@/portainer/access-control/types';
+import { DockerContainer } from '@/docker/containers/types';
+
+import { useNetwork, useDeleteNetwork } from '../queries';
+import { isSystemNetwork } from '../network.helper';
+import { useContainers } from '../../containers/queries';
+import {
+  DockerNetwork,
+  NetworkContainer,
+  NetworkResponseContainer,
+} from '../types';
+
+import { NetworkDetailsTable } from './NetworkDetailsTable';
+import { NetworkOptionsTable } from './NetworkOptionsTable';
+import { NetworkContainersTable } from './NetworkContainersTable';
+
+export function NetworkDetailsView() {
+  const router = useRouter();
+  const queryClient = useQueryClient();
+
+  const [filteredNetworkContainers, setFilteredNetworkContainers] = useState(
+    [] as NetworkContainer[]
+  );
+  const {
+    params: { id: networkId, nodeName },
+  } = useCurrentStateAndParams();
+  const environmentId = useEnvironmentId();
+
+  const networkQuery = useNetwork(environmentId, networkId);
+  const deleteNetworkMutation = useDeleteNetwork();
+  const filters = {
+    network: [networkId],
+  };
+  const containersQuery = useContainers(environmentId, filters);
+
+  useEffect(() => {
+    if (networkQuery.data && containersQuery.data) {
+      setFilteredNetworkContainers(
+        filterContainersInNetwork(networkQuery.data, containersQuery.data)
+      );
+    }
+  }, [networkQuery.data, containersQuery.data]);
+
+  return (
+    <>
+      <PageHeader
+        title="Network details"
+        breadcrumbs={[
+          { link: 'docker.networks', label: 'Networks' },
+          {
+            link: 'docker.networks.network',
+            label: networkQuery.data?.Name || '',
+          },
+        ]}
+      />
+      {networkQuery.data && (
+        <>
+          <NetworkDetailsTable
+            network={networkQuery.data}
+            onRemoveNetworkClicked={onRemoveNetworkClicked}
+          />
+
+          <AccessControlPanel
+            onUpdateSuccess={() =>
+              queryClient.invalidateQueries([
+                'environments',
+                environmentId,
+                'docker',
+                'networks',
+                networkId,
+              ])
+            }
+            resourceControl={networkQuery.data.Portainer?.ResourceControl}
+            resourceType={ResourceControlType.Network}
+            disableOwnershipChange={isSystemNetwork(networkQuery.data.Name)}
+            resourceId={networkId}
+          />
+        </>
+      )}
+      {networkQuery.data?.Options && (
+        <NetworkOptionsTable options={networkQuery.data.Options} />
+      )}
+      {filteredNetworkContainers && (
```

Comment:
I don't see a reason why `filteredNetworkContainers` will ever be undefined, as the default value is `[]` (https://github.com/portainer/portainer/pull/6700/files#diff-79d83f1faea16eaea8e38d9d4282021682c7700c86852736b2a9a1aa0136a301R29). so just remove this check